pkgs/applications/misc/oneko/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchurl, xorg, x11 }:
+
+let version = "1.2.sakura.5";
+    vname = "1.2.5";
+in stdenv.mkDerivation rec {
+  name = "oneko-" + vname;
+  src = fetchurl {
+    url = "http://www.daidouji.com/oneko/distfiles/oneko-"
+          + version + ".tar.gz";
+    sha256 = "2c2e05f1241e9b76f54475b5577cd4fb6670de058218d04a741a04ebd4a2b22f";
+  };
+  buildInputs = [ xorg.imake xorg.gccmakedep x11 ];
+  imake = xorg.imake;
+  gccmakedep = xorg.gccmakedep;
+  
+  builder = builtins.toFile "builder.sh" ''
```

Comment:
instead of creating your own builder, you may want to look at how `pkgs/tools/X11/xbrightness/default.nix` uses `imake`

Reply:
I made the changes you suggested. Using the provided builder instead of writing one manually did simplify things a lot!